index.src.html

```diff
@@ -276,32 +285,58 @@ <h4 id="shortlinks">Navigation</h4>
 <section>
   <h2 id="framework">Framework</h2>
 
-  An IPv4 address is a <dfn export local-lt="private">private address</dfn> if
-  it matches the <a>private address space</a> defined in Section 3 of
-  [[!RFC1918]], a <dfn export local-lt="local">local address</dfn> if it matches
-  the "loopback" space (`127.0.0.0/8`) defined in section 3.2.1.3 of
-  [[!RFC1122]] or the "link-local" space (`169.254.0.0/16`) defined in
-  [[!RFC3927]], and a <dfn export local-lt="public">public address</dfn>
-  otherwise.
-
-  An IPv6 address is a <a>local address</a> if it matches the "Unique Local
-  Address" prefix (`fc00::/7`) defined in Section 3 of [[!RFC4193]] or the
-  "link-local" prefix (`fe80::/10`) defined in section 2.5.6 of [[!RFC4291]],
-  and a <a>public address</a> otherwise.
-
-  ISSUE(mikewest/cors-rfc1918#3): What does "private" mean in the context of
-  IPv6?
+  <h3 id="address-space-heading">Address Space</h3>
+
+  Every IP address belongs to an IP <dfn export>address space</dfn>, which can
+  be one of three different values:
+
+  1.  <dfn export>local</dfn>: contains the local host only. In other words,
+      addresses whose target differs for every device.
+  1.  <dfn>private</dfn>: contains addresses that have meaning only within the
+      current network. In other words, addresses whose target differs based on
+      network position.
+  1.  <dfn>public</dfn>: contains all other addresses. In other words, addresses
+      whose target is the same for all devices globally on the IP network.
+
+  The <a>address space</a> of a given IP address (|address|) is determined
+  in accordance with the IANA Special-Purpose Address Registries
+  ([[IPV4-REGISTRY]] and [[IPV6-REGISTRY]]):
+
+  1.  If |address| is an IPv4 address in the "Loopback" address block
+      (`127.0.0.1/8` at time of writing), then its <a>address space</a> is
+      <a>local</a>.
+  1.  If |address| is an IPv6 address in the "Loopback Address" address block
+      (`::1/128` at time of writing), then its <a>address space</a> is
+      <a>local</a>.
+  1.  If |address| is an IPv6 address in the "IPv4-mapped Address" address block
+      (`::ffff:0:0/96` at time of writing), then its <a>address space</a> is
+      that of its embedded IPv4 address.
```

Comment:
I see that the CL ended up being dropped because it's not a great idea for Chrome to support these IP addresses in general. Given that https://crbug.com/1000740 has gone nowhere since, it seems like a good idea to cover all bases and prevent a trivial bypass of CORS-RFC1918 through mapped addresses.

I could see an argument for not mentioning them in the spec because the IETF recommends against using them apart from specific scenarios that do not apply to web browsers. That, however, would open other browser developers to the same security vulnerability if they fail to think of this case. It does not seem a valuable trade-off for theoretical purity?

Reply:
I base my above comment on an assumption that adding an ISSUE here means filing an issue in the spec repo. I don't believe we would reach a different conclusion in such an issue, hence my stance. If ISSUEs can also be peppered in specs to indicate potential pitfalls, then I agree with your suggestion.

Reply:
How about we compromise on something like: "ISSUE: Remove this once we remove support for IPv4-mapped addresses from Fetch." or "ISSUE: Consider removing this categorization, and blocking access to IPv4-mapped addresses entirely." or similar?

Reply:
That's fair! Filed issue #36 to track removing this in the longer term and added an ISSUE callout.